### PR TITLE
feat(apps): transpile via django

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

We should not be transpiling site and frontend apps in the plugin server.

## Changes

- Moves this to django, and a new `transpiler` service. It's a Node.js CLI app that communicates via stdin/stdout.
- No changes to the plugin server yet. It will ignore all transpiled code, so effectively this will disable transpilation on it. If it works well, all related code in the plugin server can be deleted.

## How did you test this code?

Updated the plugin transpilation tests.

### CODE REVIEW

**Comment by mariusandra** [comment]
Could someone review this now please? @tiina303 @xvello @Twixes @timgl

File: posthog/api/plugin.py [hunk 1]
```diff
@@ -358,10 +378,31 @@ def update_source(self, request: request.Request, **kwargs):
...
             if key not in sources:
                 performed_changes = True
                 sources[key], created = PluginSourceFile.objects.update_or_create(
-                    plugin=plugin, filename=key, defaults={"source": value}
+                    plugin=plugin,
+                    filename=key,
+                    defaults={
+                        "source": value,
+                        "transpiled": transpiled,
+                        "status": status,
```
Comment (tiina303) [inline posthog/api/plugin.py:403]: what happens if someone updates? Doesn't it need to get transpiled again?

File: posthog/api/plugin.py [hunk 2]
```diff
@@ -358,10 +378,31 @@ def update_source(self, request: request.Request, **kwargs):
...
+                        "error": error,
+                    },
                 )
             elif sources[key].source != value:
```
Comment (tiina303) [inline posthog/api/plugin.py:407]: don't we need to check if status/transpiled/error got updated too, not only value?

File: transpiler/package.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    "version": "1.0.0",
+    "description": "Transpiles site apps TSX to browser JS via stdin/stdout",
+    "main": "transpile.mjs",
+    "scripts": {
+        "test": "echo \"Error: no test specified\" && exit 1",
+        "build": "tsc -b && node build.mjs",
+        "start:dist": "node dist/index.js",
+        "start": "npm run build && npm run start:dist"
+    },
+    "author": "PostHog <hey@posthog.com>",
```
Comment (tiina303) [inline transpiler/package.json:12]: that is not a valid email anymore - as in it gets an autoreply of use some other email depending on your need

**Review by tiina303 — COMMENTED** [review]
don't know much about the transpiring side

File: posthog/api/plugin.py (hunk 1 above)
Comment (mariusandra) [reply posthog/api/plugin.py:403]: We do transpile every inputted source at lines 386 and 389

File: posthog/api/plugin.py (hunk 2 above)
Comment (mariusandra) [reply posthog/api/plugin.py]: Makes sense, thanks

File: transpiler/package.json (hunk 3 above)
Comment (mariusandra) [reply transpiler/package.json]: hmm... I removed the email completely.

**Review by mariusandra — COMMENTED** [review]
Thanks @tiina303 , ready for another look

File: production.Dockerfile [hunk 4]
```diff
@@ -25,6 +25,15 @@ FROM node:18.12.1-bullseye-slim AS frontend-build
 WORKDIR /code
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
+COPY transpiler/ transpiler/
+WORKDIR /code/transpiler
+RUN ls && corepack enable && pnpm --version && \
```
Comment (Twixes) [inline production.Dockerfile:30]: What's the `pnpm --version` for?

File: production.Dockerfile [hunk 5]
```diff
@@ -25,6 +25,15 @@ FROM node:18.12.1-bullseye-slim AS frontend-build
...
+    mkdir /tmp/pnpm-store && \
+    pnpm install --frozen-lockfile --store-dir /tmp/pnpm-store && \
+    pnpm build && \
+    rm -rf /tmp/pnpm-store
+
+WORKDIR /code
 COPY package.json pnpm-lock.yaml ./
 RUN corepack enable && pnpm --version && \
```
Comment (Twixes) [inline production.Dockerfile:38]: Now we `corepack enable` etc. twice in one stage, which seems wasteful. I think it'd be pretty nice if `transpiler-build` was a separate stage.

File: transpiler/README.md
Comment (Twixes) [inline transpiler/README.md:1]: Would be good to have a quick description here. And might be worth renaming to `plugin-transpiler`

**Review by Twixes — COMMENTED** [review]
Some monorepo considerations, but overall this makes sense!

**Review by tiina303 — APPROVED** [review]
leaving the actual transpilation code for your team to review, api side looks good

**Pull request merged**